### ISSUE TITLE
False Negatives of rules related to package manager cache clearing

### ISSUE THREAD

**Issue by Amaindex** [hadolint/hadolint#957]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
After installing software using a package manager in the RUN command, developers should clear the cache to avoid unnecessary image sizes. The following rules all involve this logic:
- [DL3032](https://github.com/hadolint/hadolint/wiki/DL3032): yum clean all missing after yum command.
- [DL3036](https://github.com/hadolint/hadolint/wiki/DL3036): zypper clean missing after zypper use.
- [DL3040](https://github.com/hadolint/hadolint/wiki/DL3040): dnf clean all missing after dnf command.
- [DL3060](https://github.com/hadolint/hadolint/wiki/DL3060): yarn cache clean missing after yarn install was run.
- etc.

### Actual behavior
However, when there is a cache clearing action before the installation action in the same RUN command, the detection of these rules will present **False Negative**.

### Steps to reproduce the behavior
Just prepare the following Dockerfile and apply Hadolint on it.
```Dockerfile
From CentOS:7.0
RUN yum clean all && yum install -y git-2.0
```
There is no output, even though it violates DL3032.

**Output of `hadolint --version`**

```bash
Haskell Dockerfile Linter 2.12.1
```
**The relevant Dockerfile**
There is a [real-world Dockerfile here](https://github.com/siaorg/sia-task/blob/f0bb2c4fd40b752bbd571e17232db7c24ad041c4/sia-task-docker/scheduler-docker/scheduler/Dockerfile#L12) that can attest to this issue.
